### DISCUSSION TITLE
How to load and resize an image on window resize?

### DISCUSSION THREAD

**Discussion by ghost** [TomSchimansky/CustomTkinter#1888] [Q&A]
Hello all,

I was looking at image_example.py and I have a question about resizing the image after the window is resized. The example code is as follows:

        self.large_test_image = customtkinter.CTkImage(Image.open(os.path.join(image_path, "pirates.jpg")), size=(500, 300))
        self.home_frame_large_image_label = customtkinter.CTkLabel(self.home_frame, text="", image=self.large_test_image)
        self.home_frame_large_image_label.grid(row=0, column=0, padx=20, pady=10)

When I maximize the window, the image is still 500,300. How is it possible to automatically resize the image to fill the entire label on Window resize?

Thanks!

**Comment by Adarsh-365** [comment]
try one thing add binding of configure like  window.bind("<Configure>",self.resize_image)   ,it will calll self.resize_image function whenever the window is resized, the next thing is to create that function and give event as argument width = event.width height = event.height which will give window width and height finally create a new image and  resize it as you wish scale using (width,height) and use  self.home_frame_large_image_label.configure to change the image

**Reply by ghost** [reply]
How do I change the actual image size though? CTkImage doesn't have a resize function

**Reply by ghost** [reply]
I did the following:

    def __init__(self):
        super().__init__()
        self.geometry("700x450")
        # set grid layout 1x2
        self.grid_rowconfigure(0,weight=1)
        self.grid_columnconfigure(1,weight=1)
        # create home frame
        self.home_frame = customtkinter.CTkFrame(self,corner_radius=0,fg_color="transparent")
        self.home_frame.grid_columnconfigure(0,weight=1)
        self.pirates = Image.open(os.path.join(image_path,"pirates.jpg"))
        self.piratesctk = customtkinter.CTkImage(self.pirates,size=(500,300))
        self.homelbl = customtkinter.CTkLabel(self.home_frame,text="",image=self.piratesctk)
        self.homelbl.grid(row=0,column=0,padx=10,pady=10)
        self.homelbl.bind('<Configure>',self._resize_image)

    def _resize_image(self,event):
        new_width = event.width
        new_height = event.height
        pilimg = self.pirates.resize((new_width,new_height)) 
        self.homelbl.configure(image=customtkinter.CTkImage(pilimg))

The resize gets called several times at app start, but eventually the width and height is 20x20. Any idea why it resizes that small?

**Reply by Adarsh-365** [reply]
instead of self.homelbl.bind('<Configure>',self._resize_image) if you create obj of window like window=CTk() then use window.bind('<Configure>',self._resize_image) might work


as you use .bind('<Configure>' to label hence  width and height is 20x20. try using window obj

**Comment by RobertCordingly** [comment]
Was there ever a solution found for this?

**Comment by Akascape** [comment]
Solution: https://github.com/TomSchimansky/CustomTkinter/discussions/2212#discussioncomment-8171509
